webapp/src/editortoolbar.tsx

```diff
@@ -136,6 +136,63 @@ export class EditorToolbar extends data.Component<ISettingsProps, {}> {
         return saveInput;
     }
 
+    protected onHwItemClick = () => {
+        this.props.parent.showChooseHwDialog(true);
+    }
+
+    protected getCompileButton(view: View, collapsed?: boolean): JSX.Element[] {
+        const targetTheme = pxt.appTarget.appTheme;
+        const { compiling, isSaving } = this.props.parent.state;
+        const compileLoading = !!compiling;
+        const compileTooltip = lf("Download your code to the {0}", targetTheme.boardName);
+        const downloadIcon = targetTheme.downloadIcon || "download";
+        const downloadText = targetTheme.useUploadMessage ? lf("Upload") : lf("Download");
+        const boards = pxt.appTarget.simulator && !!pxt.appTarget.simulator.dynamicBoardDefinition;
+
+        let downloadButtonClasses = boards ? "left attached " : "";
+        let hwIconClasses = "";
+        let hwMenuClasses = ""
+        let displayRight = false;
+        if (isSaving) {
+            downloadButtonClasses = "disabled ";
+        } else if (compileLoading) {
+            downloadButtonClasses = "loading disabled ";
+        }
+        switch (view) {
+            case View.Mobile:
+                downloadButtonClasses += "download-button-full";
+                displayRight = collapsed;
+                break;
+            case View.Tablet:
+                downloadButtonClasses += `download-button-full ${!collapsed ? 'large fluid' : ''}`;
+                hwMenuClasses = !collapsed ? "large" : "";
+                hwIconClasses = !collapsed ? "large" : "";
+                displayRight = collapsed;
+                break;
+            case View.Computer:
+            default:
+                downloadButtonClasses += "huge fluid";
+                hwMenuClasses = "large";
+                hwIconClasses = "large";
+        }
+
+        let el = [];
+        el.push(<EditorToolbarButton key="downloadbutton" role="menuitem" icon={downloadIcon} className={`primary download-button ${downloadButtonClasses}`} text={view != View.Mobile ? downloadText : undefined} title={compileTooltip} onButtonClick={this.compile} view='computer' />)
+
+        let deviceName = pxt.hwName || lf("device");
```

Comment:
Capitalize?

Reply:
it's used in "Download to {0}" below so "device" is part of the sentence